### DISCUSSION TITLE
Http client to fetch collection items from a remote source

### DISCUSSION THREAD

**Discussion by michellaurent** [tighten/jigsaw#710] [Q&A] [answered]
I would like to use the [Laravel Http client](https://laravel.com/docs/11.x/http-client) to fetch collection items from a remote API.

The docs gives an exemple with file_get_contents but I need to gives an API key to fetch the data. 

```php 
Http::withHeaders([
  "X-API-Key" => "MyApiKey"
])
  ->get("https://remote.provider.net/api/v1/forms/".$id."/submissions")
  ->json();
`

**Comment by bakerkretzmar** [comment]
The `Http` component can be installed on its own with Composer, have you tried that and did it work? It does have some dependencies on other Illuminate components and you won't be able to use it as a Facade like in your example, but it may still work. If not, or if you're getting weird errors, it's probably better to use Guzzle directly instead.

**Reply by michellaurent** [reply]
I added `illuminate/http` via composer without problems.
Is there a way to use it without the Facade (I don't know very well this concept in Laravel) or is possible to add the facade in Jigsaw ?

**Reply by michellaurent** [reply, marked as answer]
When looking into the `torchlight/torchlight-jigsaw` package, I found the missing piece of cake to be able to use the `Http` facade.

```php
// Set the root if it doesn't exist yet.
if (!Facade::getFacadeApplication()) {
    Facade::setFacadeApplication($this->container);
}
```

Adding this into `bootstrap.php` allows to use the Laravel Facade directly.

**Reply by bakerkretzmar** [reply]
Oh awesome! Didn't know that, but it looks great!
